### PR TITLE
Fix retrieval of pflag stringSlice

### LINKED ISSUE #112: BindPFlag don't work properly with pflag.StringSlice
An open and close bracket are added to the first and last item of a StringSlice when it's updated by a flag.

Here's some test code that demonstrates the issue:
```
package main

import (
	"fmt"

	jww "github.com/spf13/jwalterweatherman"
	"github.com/spf13/pflag"
	"github.com/spf13/viper"
)

func main() {
	jww.SetStdoutThreshold(jww.LevelDebug)

	viper.SetConfigName("config")
	viper.AddConfigPath(".")
	err := viper.ReadInConfig()
	if err != nil {
		panic(fmt.Errorf("fatal error: %s \n", err))
	}

	pflag.StringSlice("foo", []string{}, "Test flag for StringSlice")
	err = viper.BindPFlag("foo", pflag.Lookup("foo"))
	if err != nil {
		panic(fmt.Errorf("fatal error: %s \n", err))
	}
	pflag.Parse()

	for _, value := range viper.GetStringSlice("foo") {
		fmt.Println(value)
	}
}
```

And here's the output:
```
(04:14:41) brendan @ ~/workspace/golang/src/github.com/b0d0nne11/test 
→ go run test.go 
INFO: 2015/09/03 Trying to resolve absolute path to .
INFO: 2015/09/03 adding /home/brendan/workspace/golang/src/github.com/b0d0nne11/test to paths to search
INFO: 2015/09/03 Attempting to read in config file
INFO: 2015/09/03 Searching for config in  [/home/brendan/workspace/golang/src/github.com/b0d0nne11/test]
DEBUG: 2015/09/03 Searching for config in  /home/brendan/workspace/golang/src/github.com/b0d0nne11/test
DEBUG: 2015/09/03 Checking for /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.json
DEBUG: 2015/09/03 Checking for /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.toml
DEBUG: 2015/09/03 Checking for /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.yaml
DEBUG: 2015/09/03 Found:  /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.yaml
DEBUG: 2015/09/03 ToStringSliceE called on type: []interface {}
DEBUG: 2015/09/03 ToStringE called on type: string
DEBUG: 2015/09/03 ToStringE called on type: string
bar
baz

(04:14:52) brendan @ ~/workspace/golang/src/github.com/b0d0nne11/test 
→ go run test.go --foo='bar baz'
INFO: 2015/09/03 Trying to resolve absolute path to .
INFO: 2015/09/03 adding /home/brendan/workspace/golang/src/github.com/b0d0nne11/test to paths to search
INFO: 2015/09/03 Attempting to read in config file
INFO: 2015/09/03 Searching for config in  [/home/brendan/workspace/golang/src/github.com/b0d0nne11/test]
DEBUG: 2015/09/03 Searching for config in  /home/brendan/workspace/golang/src/github.com/b0d0nne11/test
DEBUG: 2015/09/03 Checking for /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.json
DEBUG: 2015/09/03 Checking for /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.toml
DEBUG: 2015/09/03 Checking for /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.yaml
DEBUG: 2015/09/03 Found:  /home/brendan/workspace/golang/src/github.com/b0d0nne11/test/config.yaml
DEBUG: 2015/09/03 ToStringE called on type: string
DEBUG: 2015/09/03 ToStringSliceE called on type: string
[bar
baz]
```

### PR DESCRIPTION
**Pull request by moorereason**
With some extra trace log formatting changes thrown into the mix.

Fixes #112

### CODE REVIEW

**Comment by awishformore** [comment]
Don't see any issues with this, thank you for your contribution.

**Comment by aarondl** [comment]
@awishformore @moorereason 

This PR seems to fix #200  (haven't tested)

But note that I changed two hunks in my workaround, one in `find()` as in this PR (pretty much the same code) and one in `Get()` . I wonder if the second one in `Get()` is necessary. Can anyone who knows the code base advise on this? The code I put in that case is probably wrong, but the fact that there's a similar type check going on there makes me suspicious. I just want to make sure we covered all the possibilities.

If it's truly done, then we should probably close #200 as well.

**Comment by awishformore** [comment]
I'll try to take a look soon.

File: viper.go [hunk 1]
```diff
@@ -749,45 +749,48 @@ func (v *Viper) find(key string) interface{} {
...
 	if exists && flag.HasChanged() {
-		jww.TRACE.Println(key, "found in override (via pflag):", flag.ValueString())
+		jww.TRACE.Printf("%q found in pflag override (%s): %s", key, flag.ValueType(), flag.ValueString())
 		switch flag.ValueType() {
 		case "int", "int8", "int16", "int32", "int64":
 			return cast.ToInt(flag.ValueString())
 		case "bool":
 			return cast.ToBool(flag.ValueString())
+		case "stringSlice":
+			s := strings.TrimPrefix(flag.ValueString(), "[")
```
Comment (shenwei356) [inline viper.go]: This wrongly trims the "[" and "]" in the flags. 

Fix it by:

    s := flag.ValueString()
    return s[1:len(s)-1]

Comment (shenwei356) [inline viper.go]: see https://github.com/spf13/pflag/pull/98

Comment (moorereason) [inline viper.go]: @shenwei356, your solution is probably faster, but `strings.TrimPrefix` only removes a single "[" character. The problem with pflag was that it used `strings.Trim` which would remove multiples.

Comment (shenwei356) [inline viper.go]: you are right!

**Pull request merged**